app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/sdk/ManagedCloudSdkService.java

```diff
@@ -356,53 +357,40 @@ public void onSuccess(ManagedSdkJobResult result) {
     }
 
     @Override
-    public void onFailure(Throwable t) {
+    public void onFailure(@NotNull Throwable throwable) {
       boolean jobCancelled = false;
-      if (t instanceof InterruptedException || t instanceof CancellationException) {
+      if (throwable instanceof InterruptedException || throwable instanceof CancellationException) {
         logger.info("Managed Google Cloud SDK install/update cancelled.");
         jobCancelled = true;
 
         ManagedCloudSdkServiceUiPresenter.getInstance().notifyManagedSdkJobCancellation(jobType);
       } else {
-        logger.warn("Error while installing/updating managed Cloud SDK", t);
+        logger.warn("Error while installing/updating managed Cloud SDK", throwable);
 
         ManagedCloudSdkServiceUiPresenter.getInstance()
-            .notifyManagedSdkJobFailure(jobType, t.toString());
+            .notifyManagedSdkJobFailure(jobType, throwable.toString());
       }
 
       // update status of the SDK and generate corresponding metric events.
       switch (jobType) {
         case INSTALL:
           updateStatus(SdkStatus.NOT_AVAILABLE);
 
-          FluentTrackingEventWithMetadata trackingEvent;
-          if (jobCancelled) {
-            trackingEvent =
-                UsageTrackerProvider.getInstance()
-                    .trackEvent(GctTracking.MANAGED_SDK_CANCELLED_INSTALL);
-          } else {
-            trackingEvent =
-                UsageTrackerProvider.getInstance()
-                    .trackEvent(GctTracking.MANAGED_SDK_FAILED_INSTALL)
-                    .addMetadata(GctTracking.METADATA_MESSAGE_KEY, t.toString());
-          }
-          trackingEvent.ping();
+          sendManagedSdkTrackingEvent(
```

Comment:
not a big deal. but I was thinking something like:

```
if (jobCancelled) {
  trackCancelledEvent();
} else {
  trackFailedEvent();
}
```

Might be cleaner than passing nulls around?

Reply:
We also need to distinguish between updates and installs, so something like:
```
if (jobCancelled) {
  trackCancelledEvent(INSTALL/UPDATE);
} else {
  trackFailedEvent(INSTALL/UPDATE, Throwable);
}
```
?

Reply:
ah ok. what you had is fine then.

Reply:
or your suggestion :) up to you